nixui/state_model.py

```diff
@@ -60,8 +69,15 @@ def persist_updates(self):
         self.slotmapper('changes_saved')(save_path)
 
     def undo(self, *args, **kwargs):
+        if not self.update_history:
+            self.slotmapper('no_updates_exist')()
+            logger.error('Reached unexpected branch point, attempted to undo when no update history exists')
+
         last_update = self.update_history.pop()
         self.current_values[last_update.option] = last_update.old_value
 
+        if not self.update_history:
+            self.slotmapper('no_updates_exist')()
+
         self.slotmapper('undo_performed')(last_update.option, last_update.old_value, last_update.new_value)
-        self.slotmapper(('update_field', last_update.option))(last_update.old_value)
+        self.slotmapper(('update_field', last_update.option))()
```

Comment:
Is it possible this change accidentally removed the undoing functionality?

Reply:
Yep, my bad. I was going to mention that I think the widgets should load from the model instead, and turns out that this change does *exactly* that.